### DISCUSSION TITLE
Replacing global with environment variable at compile time?

### DISCUSSION THREAD

**Discussion by joakimgunst** [swc-project/swc#5941] [Q&A] [answered]
Using `@swc/cli`, is there a way to replace a value in the code, e.g. `__BUILD_NUMBER__`, with the value of an environment variable at the time of compilation?  We would like to replace the build number in the code at compile time. If I understand correctly, `jsc.transform.optimizer.globals` only supports static values. I suppose writing an SWC plugin would work, but I'm wondering if there is a built-in solution or an existing plugin for this use case. With Webpack, [DefinePlugin](https://webpack.js.org/plugins/define-plugin/) could be used.

**Comment by kwonoj** [comment, marked as answer]
as define **plugin** does, swc plugin / or programmatic interface (`swc.transform*`) is the way to go. Cli does not have built in support.
